src/Autofac.Multitenant/MultitenantContainer.cs

```diff
@@ -394,25 +404,43 @@ private ILifetimeScope CreateTenantScope(object tenantId, Action<ContainerBuilde
     /// <returns><c>true</c> if an existing configuration was removed; otherwise, <c>false</c>.</returns>
     /// <seealso cref="ConfigurationActionBuilder"/>
     /// <seealso cref="ConfigureTenant(object, Action{ContainerBuilder})"/>
+    [SuppressMessage("CA1513", "CA1513", Justification = "ObjectDisposedException.ThrowIf is not available in all target frameworks.")]
     public bool ReconfigureTenant(object tenantId, Action<ContainerBuilder> configuration)
     {
         if (configuration == null)
         {
             throw new ArgumentNullException(nameof(configuration));
         }
 
+        if (_isDisposed == 1)
+        {
+            throw new ObjectDisposedException(nameof(ApplicationContainer));
+        }
+
         tenantId ??= _defaultTenantId;
 
-        var removed = false;
-        if (_tenantLifetimeScopes.TryRemove(tenantId, out var tenantScope))
+        var lifetimeScope = ApplicationContainer.BeginLifetimeScope(TenantLifetimeScopeTag, configuration);
```

Comment:
Might be worth a unit test for `ReconfigureTenant` on a tenant that wasn't configured (IIRC there's already one to handle a tenant that was configured) so we don't regress the behavior.

Reply:
ok